src/components/ui/SkuSelector/SkuSelector.tsx

```diff
@@ -0,0 +1,169 @@
+        disabled={disabled}
+        checked={label === selectedOption}
+        {...otherProps}
+      >
+        {variant === 'size' && (
+          // TODO: Remove this inline style. Added just to skip "Tap Target" error (Lighthouse)
+          <span style={{ padding: 48 }}>{option.label}</span>
+        )}
+        {variant === 'color' && (
+          <div style={{ backgroundColor: value, height: 40, width: 40 }} />
+        )}
+      </RadioOption>
+    )
+  }
+
+  if (variant === 'image' && 'src' in option) {
+    const { label, src, alt, disabled }: ImageSkuProps = option
+
+    return (
+      <RadioOption
+        label={label}
+        value={label}
+        disabled={disabled}
+        checked={label === selectedOption}
+        {...otherProps}
+      >
+        <Image src={src} alt={alt} variant="product.sku" />
+      </RadioOption>
+    )
+  }
+
+  return <div />
+}
+
+type DefaultSkuProps = {
+  /**
+   * Label to describe the SKU when selected.
+   */
+  label: string
+  /**
+   * Current value for this SKU.
+   */
+  value: string
+  /**
+   * Specifies that this option should be disabled.
+   */
+  disabled?: boolean
+}
+
+interface ImageSkuProps {
+  /**
+   * Alternative text description of the image.
+   */
+  alt: string
+  /**
+   * Image URL.
+   */
+  src: string
+  /**
+   * Label to describe the image when selected.
+   */
+  label: string
+  /**
+   * Specifies that this option should be disabled.
+   */
+  disabled?: boolean
+}
+
+type ImageVariant = 'image'
+
+type Sku<V> = V extends ImageVariant ? ImageSkuProps : DefaultSkuProps
+
+type Variant = 'color' | 'size' | 'image'
+
+export interface SkuSelectorProps {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress,
+   * testing-library, and jest).
+   */
+  testId?: string
+  /**
+   * Specify which variant the component should handle.
+   */
+  variant: Variant
+  /**
+   * SKU options that should be rendered.
+   */
+  options: Array<Sku<Variant>>
+  /**
+   * Default SKU option.
+   */
+  defaultSku?: string
+  /**
+   * Section label for the SKU selector.
+   */
+  label?: string
+  /**
+   * Function to be triggered when SKU option change.
+   */
+  onChange?: ChangeEventHandler<HTMLInputElement>
+}
+
+const SkuSelector = ({
+  label,
+  variant,
+  options,
+  onChange,
+  defaultSku,
+  testId = 'store-sku-selector',
+}: SkuSelectorProps) => {
+  const [selectedSku, setSelectedSku] = useState<string>(defaultSku ?? '')
+
+  return (
+    <div data-store-sku-selector data-testid={testId} data-variant={variant}>
+      {label && (
+        <Label data-sku-selector-label>
+          {label}: {selectedSku}
+        </Label>
+      )}
+      <RadioGroup
+        selectedValue={selectedSku}
+        name={`sku-selector-${variant}`}
+        onChange={(e) => {
+          onChange?.(e)
+          setSelectedSku(e.currentTarget.value)
+        }}
+      >
+        {options.map((option, index) => {
+          return (
+            <Option
```

Comment:
nit: I prefer personally to not have this component, but rather have the if's etc in a ternary clause in here. I think it's easier to reason about. Something like:
```suggestion
            <RadioOption>{variant === "image" ? <Image/> : <><span> .... </>}</RadioOption>
```

but this is just a personal preference. If  you want you can keep like this :)

Reply:
@tlgimenes it's a good point, but this probably would make the render body very inflated, as I would still need to do the same type narrowing and we have more than two conditions up there, but I'm looking at a more optimized way of this without having the `<Option />`